paddle/fluid/platform/device_context.cc

```diff
@@ -408,6 +412,14 @@ void set_cur_input_shape_cache_capacity(int input_shape_cache_capacity) {
   cur_input_shape_cache_capacity = input_shape_cache_capacity;
 }
 
+void set_cur_paddle_data_layout(framework::DataLayout dl) {
+  cur_paddle_data_layout = dl;
+}
+
+framework::DataLayout get_cur_paddle_data_layout(void) {
+  return cur_paddle_data_layout;
+}
+
```

Comment:
There are too many global functions, should line391-421 in Class MKLDNNDeviceContext? See the duplicated issue #18944

Reply:
we can move all these global variables and functions into Class MKLDNNDeviceContext and set as static firstly, user can still use as global func.

Reply:
Ok, I have to think about it. I would like to do this little refactoring in next PR (NHWC for conv and conv_transpose).